packages/manager/apps/telecom/src/app/telecom/telephony/line/assist/rma/rma.controller.js

```diff
@@ -1,101 +1,159 @@
-      .fetchRma()
+export default class TelecomTelephonyLineAssistRmaCtrl {
+  constructor($translate, lineAssistRmaService, TucToast, TucToastError) {
+    this.$translate = $translate;
+    this.lineAssistRmaService = lineAssistRmaService;
+    this.TucToast = TucToast;
+    this.TucToastError = TucToastError;
+  }
+
+  $onInit() {
+    this.rmaList = null;
+    this.modalPrimaryAction = null;
+    this.modalTitle = null;
+    this.modalContent = null;
+    this.modalRma = null;
+    this.isOpenModal = false;
+    this.actionDoing = false;
+
+    return this.lineAssistRmaService
+      .fetchRma(this.billingAccount, this.serviceName)
       .then((result) => {
-        self.rmaList = result;
+        this.rmaList = result;
       })
-      .catch((err) => new TucToastError(err));
+      .catch((err) => new this.TucToastError(err));
   }
 
-  self.fetchPhone = function fetchPhone() {
-    return OvhApiTelephony.Line()
-      .Phone()
-      .v6()
-      .get({
-        billingAccount: $stateParams.billingAccount,
-        serviceName: $stateParams.serviceName,
-      }).$promise;
-  };
+  modalDismiss() {
+    this.isOpenModal = false;
+    this.modalPrimaryAction = null;
+    this.modalTitle = null;
+    this.modalContent = null;
+    this.modalRma = null;
+  }
 
-  self.fetchRma = function fetchRma() {
-    return OvhApiTelephony.Line()
-      .Phone()
-      .RMA()
-      .v6()
-      .query({
-        billingAccount: $stateParams.billingAccount,
-        serviceName: $stateParams.serviceName,
-      })
-      .$promise.catch((err) => {
-        if (err.status === 404) {
-          // line has no phone
-          return [];
-        }
-        return $q.reject(err);
-      })
-      .then((rmaIds) =>
-        $q.all(
-          rmaIds.map(
-            (id) =>
-              OvhApiTelephony.Line()
-                .Phone()
-                .RMA()
-                .v6()
-                .get({
-                  billingAccount: $stateParams.billingAccount,
-                  serviceName: $stateParams.serviceName,
-                  id,
-                }).$promise,
-          ),
-        ),
-      );
-  };
+  trackDownloadPdf() {
+    this.rmaTrackClick('rma-download-receipt');
+  }
 
-  self.cancelRma = function cancelRma(rma) {
-    set(rma, 'isCancelling', true);
-    return OvhApiTelephony.Line()
-      .Phone()
-      .RMA()
-      .v6()
-      .cancel({
-        billingAccount: $stateParams.billingAccount,
-        serviceName: $stateParams.serviceName,
-        id: rma.id,
-      })
-      .$promise.then(() => {
-        remove(self.rmaList, { id: rma.id });
-        TucToast.success(
-          $translate.instant('telephony_line_assist_rma_cancel_success'),
+  generateCancelRmaModal(rma) {
+    this.rmaTrackClick('rma-cancel');
+    this.modalRma = rma;
+    this.modalTitle = TEXT_FOR_MODAL.cancelRma;
+    this.modalContent = `${TEXT_FOR_MODAL.cancelRma}_modal_content`;
+    this.modalPrimaryAction = ACTION_TYPE.cancelRma;
```

Comment:
may be we can consider to use a modal object to group properties ?
```suggestion
    this.modal = {
    rma,
    title: TEXT_FOR_MODAL.cancelRma,
    content: `${TEXT_FOR_MODAL.cancelRma}_modal_content`,
    primaryAction: ACTION_TYPE.cancelRma,
    }
```

Reply:
I fixed.